integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/VariableModesTest.kt

```diff
@@ -0,0 +1,272 @@
+        customizations.mergeFrom(LocalCustomizationContext.current)
+        var nodeName = "#stage"
+        val rootNodeQuery = NodeQuery.NodeName(nodeName)
+        if (openLinkCallback != null) customizations.setOpenLinkCallback(nodeName, openLinkCallback)
+
+        customizations.setCustomComposable { mod, name, query, parentComponents, tapCallback ->
+            CustomComponent(mod, name, query, parentComponents, tapCallback)
+        }
+
+        val (docId, setDocId) = remember { mutableStateOf("HhGxvL4aHhP8ALsLNz56TP") }
+        val queries = queries()
+        CompositionLocalProvider(LocalCustomizationContext provides customizations) {
+            DesignDoc(
+                "VariablesTestDoc",
+                docId,
+                rootNodeQuery,
+                customizations = customizations,
+                modifier = modifier.semantics { sDocClass = className },
+                serverParams = DocumentServerParams(queries, ignoredImages()),
+                setDocId = setDocId,
+                designSwitcherPolicy = DesignSwitcherPolicy.SHOW_IF_ROOT,
+                designComposeCallbacks = designComposeCallbacks,
+            )
+        }
+    }
+
+    @Composable
+    final fun CustomComponent(
+        modifier: Modifier,
+        nodeName: String,
+        rootNodeQuery: NodeQuery,
+        parentComponents: List<ParentComponentInfo>,
+        tapCallback: TapCallback?,
+    ) {
+        val customizations = remember { CustomizationContext() }
+        if (tapCallback != null) customizations.setTapCallback(nodeName, tapCallback)
+        customizations.mergeFrom(LocalCustomizationContext.current)
+        val (docId, setDocId) = remember { mutableStateOf("HhGxvL4aHhP8ALsLNz56TP") }
+        val queries = queries()
+        queries.add(nodeName)
+        CompositionLocalProvider(LocalCustomizationContext provides customizations) {
+            DesignDoc(
+                "VariablesTestDoc",
+                docId,
+                rootNodeQuery,
+                customizations = customizations,
+                modifier = modifier,
+                serverParams = DocumentServerParams(queries, ignoredImages()),
+                setDocId = setDocId,
+                designSwitcherPolicy = DesignSwitcherPolicy.SHOW_IF_ROOT,
+                parentComponents = parentComponents
+            )
+        }
+    }
+}
+
+val VariablesTestDoc2 = VariablesTestGen2()
+
+@Composable
+fun VariableModesTest() {
+    val theme = remember { mutableStateOf<Theme?>(null) }
+    val leftModeMaterial = remember { mutableStateOf(LightDarkMode.Light) }
+    val rightModeMaterial = remember { mutableStateOf(LightDarkMode.Dark) }
+    val leftModeMyTheme = remember { mutableStateOf(LightDarkMode.Light) }
+    val rightModeMyTheme = remember { mutableStateOf(LightDarkMode.Dark) }
+
+    println("### Theme ${theme.value?.name}")
+    DesignVariableCollectionProvider(theme.value?.themeName) {
```

Comment:
This method of setting a theme/collection uses a compositionLocal object. The developer wraps their composable around this, saying that all composables under this should use the specified theme. We could alternatively remove this function in favor of allowing the developer to use an annotation to specify the collection, copying how a mode is set. I've provided different implementations of setting the mode and collection to demonstrate what it would look like, but ultimately I think they should be the same.